response_operations_ui/views/collection_exercise.py

```diff
@@ -526,6 +528,9 @@ def _get_form_type(file_name):
 @collection_exercise_bp.route("/<short_name>/<period>/edit-collection-exercise-details", methods=["GET"])
 @login_required
 def view_collection_exercise_details(short_name, period):
+    if not uaa_controller.user_has_permission("surveys.edit"):
+        logger.error("User has insufficient permissions to access this page", user_id=session["user_id"])
```

Comment:
This shouldn't really be an error, and it shouldn't be a 500, it should be a 401